### PR TITLE
fix: bug toast message for when export default is not present in js object.

### PR DESCRIPTION
**Pull request by PrasadMadine**
## Description:

Fix toast message for when export default is not present in JS Object(`Start object with export default`).
`undefined` should not be present in the error message when body of JSObject is emptied and show different toast message `JS object must contain 'export default'.`.

> I have raised this PR to fix the `toast message for when export default is not present in JSObject and the JSObject is emptied`

Fixes https://github.com/appsmithorg/appsmith/issues/36185

### Before resoving bug:
image

### When there is no export default:
Screenshot from 2024-09-13 11-25-04

### When JSObject is emptied:
Screenshot from 2024-09-13 11-25-13

## Summary by CodeRabbit

- **New Features**
	- Introduced new test cases to validate error handling for empty JavaScript objects and ensure users receive appropriate feedback when the required export syntax is missing.
	- Added new error message functions to enhance clarity regarding JavaScript object export requirements.

- **Bug Fixes**
	- Enhanced error messages for JavaScript object creation to provide more specific information about errors, improving user experience during debugging.
	- Simplified error messages for parsing errors to enhance clarity, potentially improving user experience during error handling.

### CODE REVIEW

**Comment by PrasadMadine** [comment]
Hello @sagar-qa007 @ApekshaBhosale , Could you please review this PR
Thankyou

**Review by rishabhrathod01 — COMMENTED** [review]
@PrasadMadine For both the error messages could you please remove the `at JSObjeect` / `at JSObject.body`.

**Comment by PrasadMadine** [comment]
Hello @rishabhrathod01 , I have addressed the comments, could you please review this PR.
Thankyou

File: app/client/src/workers/Evaluation/JSObject/index.ts [hunk 1]
```diff
@@ -219,15 +219,28 @@ export function saveResolvedFunctionsAndJSUpdates(
...
-    };
-    dataTreeEvalRef.errors.push(errors);
+    if (entity.body.trim() !== "") {
+      const errors = {
+        type: EvalErrorTypes.PARSE_JS_ERROR,
+        context: {
+          entity: entity,
+          propertyPath: entityName + ".body",
+        },
+        message: "Start object with export default",
```
Comment (mohanarpit) [inline app/client/src/workers/Evaluation/JSObject/index.ts:229]: Is it possible to move this raw string to `messages.ts` file as well? Ideally, we would like to store all our user-facing messages in a single location.

Comment (PrasadMadine) [reply app/client/src/workers/Evaluation/JSObject/index.ts:229]: Yeah it's possible @mohanarpit , will update that as well.
Thankyou

**Comment by PrasadMadine** [comment]
Hello @sagar-qa007 , I have added the unit test cases, could you please review this PR.

File: app/client/src/workers/Evaluation/JSObject/index.ts (hunk 1 above)
Comment (PrasadMadine) [reply app/client/src/workers/Evaluation/JSObject/index.ts]: Hello @mohanarpit , I have addressed the comment, Could you please review it.

**Comment by rishabhrathod01** [comment]
@PrasadMadine could please resolve the merge conflict with the release branch on this PR?

**Comment by rishabhrathod01** [comment]
@PrasadMadine This is in our review pipeline and will be picked up soon.

**Comment by rishabhrathod01** [comment]
@PrasadMadine Please resolve the merge conflict with the latest release branch.

**Comment by rishabhrathod01** [comment]
@PrasadMadine could you please check and fix the type error here https://github.com/appsmithorg/appsmith/actions/runs/11261876012/job/31316293444?pr=36790 that is leading to build failure.

**Comment by rohan-arthur** [comment]
we will wait until 17 Oct, and move this back to the backlog if there are no further updates.

**Pull request closed without merging**